### PR TITLE
Refresh csharp topic

### PR DESCRIPTION
**Pull request by NextTurn**
### Thank you for contributing! Please confirm this pull request meets the following requirements:

- [x] I followed the contributing guidelines: https://github.com/github/explore/blob/master/CONTRIBUTING.md
- [x] I have no affiliation with the project I am suggesting (as a maintainer, creator, contractor, or employee).

### Which change are you proposing?

  - [x] Suggesting edits to an existing topic or collection
  - [ ] Curating a new topic or collection

---------------------------------------------------------------------

### Editing an existing topic or collection

I'm suggesting these edits to an existing topic or collection:
- [x] Image (and my file is `*.png`, square, dimensions 288x288)
- [x] Content (and my changes are in `index.md`)

### CODE REVIEW

**Comment by kenyonj** [comment]
Looks like you removed the lines from the PR template but didn't add any explanations. Can you please add back in an explanation to why these changes are being suggested?

File: topics/csharp/index.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-aliases: c-sharp, csharp-code, csharp7, csharp-core, csharp6, csharp-resources
+aliases: c-sharp
 created_by: Anders Hejlsberg
 display_name: C#
+github_url: https://github.com/dotnet/csharplang
 logo: csharp.png
 related: language, dotnet
-released: June 2000
-short_description: C# (C sharp) is an object-oriented programming language by Microsoft.
+released: January 2002
```
Comment (kenyonj) [inline topics/csharp/index.md]: Looks like the wiki page has 2000 as the initial release. Do you have any sources to this January 2002 date?

File: topics/csharp/index.md [hunk 2]
```diff
@@ -1,13 +1,16 @@
 ---
-aliases: c-sharp, csharp-code, csharp7, csharp-core, csharp6, csharp-resources
+aliases: c-sharp
```
Comment (kenyonj) [inline topics/csharp/index.md]: Are these aliases you are removing not typically used?

**Comment by NextTurn** [comment]
- Adds GitHub URL
- Fixes released date
- Updates website URL
- Fixes description: C# is not only "used for web development" and not "meant for Microsoft's .NET Framework"
- Some other changes

File: topics/csharp/index.md [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+short_description: C# ("C sharp") is an object-oriented and type-safe programming language.
 topic: csharp
-url: https://docs.microsoft.com/en-us/dotnet/csharp/csharp
+url: https://docs.microsoft.com/dotnet/csharp/
 wikipedia_url: https://en.wikipedia.org/wiki/C_Sharp_(programming_language)
 ---
-C# is a programming language used for web development. It is a hybrid of C and C++ that functions similar to Java and is meant for Microsoft's .NET framework.
+C# (pronounced "C sharp") is a simple, modern, object-oriented, and type-safe programming language.
+
+Its roots in the C family of languages makes C# immediately familiar to C, C++, Java, and JavaScript programmers.
```
Comment (kenyonj) [inline topics/csharp/index.md]: Should `JavaScript` be included in this list? I consider myself pretty familiar with JavaScript, but I wouldn't say the same for C#. I might even say the same for `Java`. I could be way off base, what are your thoughts?

File: topics/csharp/index.md (hunk 1 above)
Comment (NextTurn) [reply topics/csharp/index.md]: According to the wiki page, it first **appeared** in 2000, while its first **release** was in 2002.

File: topics/csharp/index.md (hunk 2 above)
Comment (NextTurn) [reply topics/csharp/index.md]: Yes, I don't think so. However, I can't count the number of related repositories now. If there is any mistake I'll add them back later.

Comment (kenyonj) [reply topics/csharp/index.md]: I am generally 👍 on not overloading the aliases. So, I like this change.

File: topics/csharp/index.md (hunk 3 above)
Comment (NextTurn) [reply topics/csharp/index.md]: C# and JavaScript are not much related.

Comment (NextTurn) [reply topics/csharp/index.md]: The text was simply taken from https://dotnet.microsoft.com/languages.

Comment (kenyonj) [reply topics/csharp/index.md]: If that's the case then I would suggest changing it to something like this:

```suggestion
Its roots in the C family of languages makes C# immediately familiar to C and C++ programmers.
```
I don't think anyone could argue with that assertion.

Comment (NextTurn) [reply topics/csharp/index.md]: Personally I think JavaScript can be removed. But C# is very similar to Java.

Comment (NextTurn) [reply topics/csharp/index.md]: /cc @leecow

**Pull request merged**